services/py-api/tests/unit_tests/test_database_layer/test_participant_repository.py

```diff
@@ -69,3 +69,53 @@ async def test_create_participant_general_exception(
     assert isinstance(result.err_value, Exception)
     # Check that the error message is the one in the Exception
     assert str(result.err_value) == "Test error"
+
+@pytest.mark.asyncio
+async def test_create_participant_random_case_success(
+    ten_sec_window: Tuple[datetime, datetime],
+    mock_input_data_random: ParticipantRequestBody,
+    repo: ParticipantsRepository,
+) -> None:
+    start_time, end_time = ten_sec_window
+
+    result = await repo.create(mock_input_data_random)
+
+    assert isinstance(result, Ok)
+    assert isinstance(result.ok_value, Participant)
+    assert result.ok_value.name == mock_input_data_random.name
+    assert result.ok_value.email == mock_input_data_random.email
+    assert result.ok_value.is_admin == mock_input_data_random.is_admin
+    assert result.ok_value.email_verified is False
+    assert result.ok_value.team_id is None
+    assert start_time <= result.ok_value.created_at <= end_time, "created_at is not within the 10-second window"
+    assert start_time <= result.ok_value.updated_at <= end_time, "updated_at is not within the 10-second window"
+
+@pytest.mark.asyncio
+async def test_create_participant_random_case_duplicate_email_error(
+    db_manager_mock: Mock, mock_input_data_random: ParticipantRequestBody, repo: ParticipantsRepository
+) -> None:
+    # Simulate a DuplicateKeyError raised by insert_one to represent a duplicate email
+    db_manager_mock.get_collection.return_value.insert_one = AsyncMock(
+        side_effect=DuplicateKeyError("Duplicate email error")
+    )
```

Comment:
As we already have written tests for the repository methods. I don't think a special case for the random participant is needed. The respository methods only communciate with the database to do create, read, update operations, they should not care if we are adding a random or admin participant. Unless you write a new method that you need in the respoitory layer, you don't need to create additional test cases for it. Thank you so much for your work man!